crates/adapters/src/transport/kafka/ft/input.rs

```diff
@@ -76,27 +79,62 @@ struct KafkaFtInputContext {
     endpoint: Mutex<Weak<KafkaFtInputReaderInner>>,
 
     deferred_logging: DeferredLogging,
+
+    // True if the sasl.mechanism is oauthbearer.
+    oauthbearer: bool,
 }
 
 impl KafkaFtInputContext {
-    fn new() -> Self {
+    fn new(oauthbearer: bool) -> Self {
         Self {
             endpoint: Mutex::new(Weak::new()),
             deferred_logging: DeferredLogging::new(),
+            oauthbearer,
         }
     }
 }
 
 impl ClientContext for KafkaFtInputContext {
+    const ENABLE_REFRESH_OAUTH_TOKEN: bool = true;
+
+    fn log(&self, level: RDKafkaLogLevel, fac: &str, log_message: &str) {
+        self.deferred_logging.log(level, fac, log_message);
+    }
+
     fn error(&self, error: KafkaError, reason: &str) {
         // eprintln!("Kafka error: {error}");
         if let Some(endpoint) = self.endpoint.lock().unwrap().upgrade() {
             endpoint.push_error(error, reason);
         }
     }
 
-    fn log(&self, level: RDKafkaLogLevel, fac: &str, log_message: &str) {
-        self.deferred_logging.log(level, fac, log_message);
+    fn generate_oauth_token(&self, _: Option<&str>) -> Result<OAuthToken, Box<dyn Error>> {
+        // TODO: Currently, OAUTHBEARER only works with AWS MSK.
+        if self.oauthbearer {
+            let region = {
+                let region = std::env::var("AWS_REGION").ok();
+                let default = std::env::var("AWS_DEFAULT_REGION").ok();
```

Comment:
I'd expect this to be in the configuration, not the environment. I don't think we provide a way to set environment variables.

Reply:
@abhizer I expect aws-config to pick these up automatically and you shouldnt have to do it yourself. You can use the S3 connector as a reference.

Reply:
We don't use `aws-config` directly, even if we did, `region` might be None. So now, the connector definition requires the `region` field when `sasl.mechanism` is `oauthbearer`.